app/templating/view_context.py

```diff
@@ -152,40 +159,46 @@ def _build_calculated_summary_section_list(schema, rendered_block, current_locat
     return [section]
 
 
-def _remove_unwanted_questions_answers(block, answers_in_block, answer_ids_to_keep):
+def _remove_unwanted_questions_answers(block, answers_in_block, answer_ids_to_keep, answer_store, metadata, schema):
+    """
+    Evaluates questions in a block and removes any questions not containing a relevant answer
+    """
+    block_question = choose_question_to_display(block, schema, answer_store, metadata)
+
     reduced_block = block.copy()
     questions_to_keep = [
         answer['parent_id'] for answer in answers_in_block.values() if answer['id'] in answer_ids_to_keep
     ]
 
     questions = []
-    for question in reduced_block['questions']:
-        if question['id'] in questions_to_keep:
-            answers_to_keep = [answer for answer in question['answers'] if answer['id'] in answer_ids_to_keep]
-            question['answers'] = answers_to_keep
-            questions.append(question)
+
+    if block_question['id'] in questions_to_keep:
+        answers_to_keep = [answer for answer in block_question['answers'] if answer['id'] in answer_ids_to_keep]
+
+        block_question['answers'] = answers_to_keep
+        questions.append(block_question)
 
     reduced_block['questions'] = questions
```

Comment:
These are all redundant?
`questions = []`
`questions.append(block_question)`
`reduced_block['questions'] = questions`

Reply:
why?

Reply:
The code is creating a subset of questions and setting `reduced_block['questions']` to that set of questions. I don't see what's redundant here?

Reply:
Unless I've misread something, there is only ever 1 question?
`questions.append(block_question)` only runs once?.

`reduced_block['questions'] = questions` -> We no longer use `questions`?
So would it not be `reduced_block['question'] = block_question or {}`?